improver/blending/spatial_weights.py

```diff
@@ -244,33 +108,33 @@ def create_template_slice(cube_to_collapse, blend_coord):
                         dimension other than the dimension associated with
                         blend_coord.
         """
-        # Takes slices over x, y coord_to_collapse
-        # Find blend dim coord associated with blend coord, sometimes we name a
-        # blend_coord that is a aux_coord associated with a dim coord rather
-        # than using the name of the dim_coord itself.
-        # Here we reset blend_coord to the name of the dim_coord to catch the
-        # case where blend_coord is an aux_coord.
-        blend_dim = cube_to_collapse.coord_dims(blend_coord)
+        # Find dimension coordinate associated with blend coord (which may be an
+        # auxiliary coordinate)
+        blend_dim = cube_to_collapse.coord_dims(self.blend_coord)
         if len(blend_dim) == 1:
             blend_dim = blend_dim[0]
         else:
             message = (
                 "Blend coordinate must only be across one dimension. "
                 "Coordinate {} is associated with dimensions {}"
             )
-            message = message.format(blend_coord, blend_dim)
+            message = message.format(self.blend_coord, blend_dim)
             raise ValueError(message)
-        blend_coord = cube_to_collapse.coord(
+        self.blend_coord = cube_to_collapse.coord(
             dimensions=blend_dim, dim_coords=True
         ).name()
         # Find original dim coords in input cube
-        original_dim_coords = [coord.name() for coord in cube_to_collapse.dim_coords]
-        # Slice over relevant coords.
+        original_dim_coords = get_dim_coord_names(cube_to_collapse)
+        # Slice over required coords
         x_coord = cube_to_collapse.coord(axis="x").name()
         y_coord = cube_to_collapse.coord(axis="y").name()
-        coords_to_slice_over = [blend_coord, y_coord, x_coord]
+        coords_to_slice_over = [self.blend_coord, y_coord, x_coord]
+        # If the dimensions are already as required, return cube unchanged
+        if original_dim_coords == coords_to_slice_over:
```

Comment:
There's a codecov flag saying this line if statement isn't tested. 
See unit test improver_tests/blending/spatial_weights/test_SpatiallyVaryingWeightsFromMask.py::Test__create_template_slice::test_basic_no_change

After printing some stuff I think it's cos your get_dim coords function classes air_temperature as a dim coord. This means this air temp coord gets removed from teh result cube, even though the test doesn't fail.

Reply:
Yes, that's because air temperature IS a dim coord... I'd forgotten that our cubes would have that.
I'll set up a test with just the three coordinates on the input cube.